src/Simplex/Messaging/Agent/Store/SQLite.hs

```diff
@@ -1041,6 +1042,20 @@ deletePendingMsgs :: DB.Connection -> ConnId -> SndQueue -> IO ()
 deletePendingMsgs db connId SndQueue {dbQueueId} =
   DB.execute db "DELETE FROM snd_message_deliveries WHERE conn_id = ? AND snd_queue_id = ?" (connId, dbQueueId)
 
+getExpiredSndMessages :: DB.Connection -> ConnId -> SndQueue -> UTCTime -> IO [InternalId]
+getExpiredSndMessages db connId SndQueue {dbQueueId} currentTs =
+  map fromOnly
+    <$> DB.query
+      db
+      [sql|
+        SELECT d.internal_id
+        FROM snd_message_deliveries d
+        JOIN messages m ON m.conn_id = d.conn_id AND m.internal_id = d.internal_id
+        WHERE d.conn_id = ? AND d.snd_queue_id = ? AND d.failed = 0 AND m.internal_ts < ?
+        ORDER BY d.internal_id ASC
+      |]
+      (connId, dbQueueId, currentTs)
```

Comment:
maybe I don't see something simple, but seems like current time gets passed all the way down here..